### PR TITLE
[XRP] Add support for Escrow transactions

### PR DESCRIPTION
**Pull request by rikublock**
## Description

Add support for the following XRP Ledger transaction types:
- `EscrowCreate`: https://xrpl.org/escrowcreate.html
- `EscrowCancel`: https://xrpl.org/escrowcancel.html
- `EscrowFinish`: https://xrpl.org/escrowfinish.html

## Types of changes

- New feature (non-breaking change which adds functionality)

## Open questions
- Running clang-format would introduce a lot of unrelated changes as the files weren't formatted properly to being with. Apply anyway?
- What are your naming conventions for class variables? The current implementation is shadowing declarations.

## Checklist
- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.
- [ ] Linting, code formatting

### CODE REVIEW

File: tests/chains/XRP/TWAnySignerTests.cpp [hunk 1]
```diff
@@ -121,6 +121,168 @@ TEST(TWAnySignerRipple, SignTokenPayment1) {
     EXPECT_EQ(hex(output.encoded()), "12000022000000002401ec61e0201b01ec61f561d48a68d1c931200000000000000000000000000055534400000000004b4e9c06f24296074f7bc48f92a97916c6dc5ea968400000000000000a73210348c331ab218ba964150490c83875b06ccad2100b1f5707f296764712738cf1ca74473045022100a938783258d33e2e3e6099d1ab68fd85c3fd21adfa00e136a67bed8fddec6c9a02206cc6784c1f212f19dc939207643d361ceaa8334eb366722cf33b24dc7669dd7a81143a2f2f189d05abb8519cc9dee0e2dbc6fa53924183148132e4e20aecf29090ac428a9c43f230a829220d");
 }
 
+TEST(TWAnySignerRipple, SignEscrowCreate) {
+    // https://testnet.xrpl.org/transactions/3F581927C742D5FAE65FB0759D0F04EF3B64B4A087911B07975816ECCB59915B
```
Comment (satoshiotomakan) [inline tests/chains/XRP/TWAnySignerTests.cpp:125]: Transaction not found. Also could you please broadcast the transactions on mainnet?
The checklist contains the step to broadcast at least one transaction on mainnet:
https://developer.trustwallet.com/developer/wallet-core/newblockchain#blockchain-checklist

**Review by satoshiotomakan — CHANGES_REQUESTED** [review]
Thank you for the contribution! Only one note

File: tests/chains/XRP/TWAnySignerTests.cpp (hunk 1 above)
Comment (rikublock) [reply tests/chains/XRP/TWAnySignerTests.cpp:125]: The transactions are definitely there. Maybe a temporary problem with the explorer? Seems to work now.
I assumed the mainnet transaction would only be necessary when adding an entirely new blockchain, but I'll see what I can do about that.

Comment (rikublock) [reply tests/chains/XRP/TWAnySignerTests.cpp:150]: It seems indeed unreliable. Depending on which backend node is selected, the transaction history might not be available. I added a bunch of main net transactions, hope that helps.

**Review by satoshiotomakan — APPROVED** [review]
LGTM, thank you for the contribution!

**Review by Milerius — APPROVED** [review]

**Pull request merged**